### PR TITLE
chore: add timestamp validation & add tests to cover

### LINKED ISSUE #480: Zoom signature check missing timestamp verification
This might allow for replay attacks

https://hookdeck.com/webhooks/guides/webhook-security-vulnerabilities-guide#timestamped-signatures

https://github.com/nearform/zoom-shuffle-bot/blob/5eb4dcccddd09de84283327b6b7a74b2330abb4f/plugins/verifyRequest.js#L5

### PR DESCRIPTION
**Pull request by ollyjohn**
Fixes #480 

### Manual testing carried out
- [x] Request with no timestamp rejected
- [x] Request with timestamp >= 5 minutes in the past rejected
- [x] Request with bad signature still rejected
- [x] Request with good signature and timestamp < 5 minutes in the past allowed

### CODE REVIEW

File: plugins/verifyRequest.js [hunk 1]
```diff
@@ -1,8 +1,15 @@
 import createError from 'http-errors'
 import { createVerificationSignature } from '../helpers/crypto.js'
 
+const TIMESTAMP_THRESHOLD = 5 * 60 * 1000 // 5 minutes
```
Comment (simoneb) [inline plugins/verifyRequest.js:4]: is there any specific reason why we're leaving so much leeway (5 mins) in the timestamp? is it recommended by the zoom docs by any chance?

Comment (simoneb) [reply plugins/verifyRequest.js:4]: works for me 👌

**Review by simoneb — APPROVED** [review]
LGTM

File: plugins/verifyRequest.js [hunk 2]
```diff
@@ -1,8 +1,15 @@
...
+
 export default async function verifyRequest(req) {
   const timestamp = req.headers['x-zm-request-timestamp']
+
+  if (Date.now() - Number(timestamp) >= TIMESTAMP_THRESHOLD) {
```
[the remark this answers is not in the document]
Comment (ollyjohn) [reply plugins/verifyRequest.js:9]: Ah yeah, nice catch!

**Review by matt-clarson — APPROVED** [review]

**Review by matt-clarson — APPROVED** [review]

**Pull request merged**